ansible/deploy_proxy.yml

```diff
@@ -96,7 +128,7 @@
         tasks_from: custom_site
       vars:
         site_config: {name: cas_ssl, vars_file: cas_ssl}
-      when: active_sites.cas_ssl == True
+      when: "'cas_ssl' in special_sites"
```

Comment:
How different are these from one another?

Reply:
I'm not sure I understand the question, so I'll answer a question I do understand, and maybe there'll be some overlap.

The impetus for the change from `active_sites` to `special_sites` was to (1) make sites whose presence could be logically determined not have to be declared and (2) to make the remaining sites declareable by simply mentioning their name in a list, rather than having to mention _all_ sites as dict keys with the value being either true or false. So

For example, for enikshay it goes from
```
active_sites:
  cchq_ssl: True
  cchq_www_redirect_secure: False
  cchq_http: True
  cchq_www_redirect_insecure: False
  cchq_http_j2me: True
  commtrack_ssl: False
  commtrack_http: False
  tableau: False
  icds_tableau: False
  wiki: False
  wiki_http: False
  motech: False
  motech2: False
  riakcs: True
  enikshay_ssl: True
```
to
```
special_sites:
  - enikshay_ssl
```
because `enikshay_ssl` is the only thing that was (2) `True` and (1) not automatically determinable.

Reply:
Offlined with @emord. To summarize:
- He left off half his question—the thing he was asking about what why we have `enikshay_ssl`, `cas_ssl`, and `pna_ssl` and why they can't share a single template (and have differences determined by variables)
- I looked at this question a little bit when doing this PR and determined they were different enough that refactoring them to share a template wasn't going to be the most straightforward thing to do and would require some individual focus
- Those files are for handling requests to "original" DNS names—so cas.commcarehq.org, etc. and my understanding is that each of these envs has a different relationship to its "original" dns name (for example, CAS still handles form submissions etc at that address, whereas maybe the other envs don't), so the template, if they did share one, would definitely have a fair bit of branching. (It's TBD in my head whether it would end up being cleaner in the end or not.)
- emord pointed out that enikshay's winding down soon and at that point there'd just be two anyway (which means either it matters less, or it's easier to refactor, or both)

That about summarizes our conversation, and it was confusing enough to me when I was looking at it earlier that I decided to document here as well.